Support XSD compilation on browser

lchanmann opened the pull request:
## Changes
- Support XSD compilation on browser
- Refactor compileAll to delegate to compile
- Activate xsd flag when loading XSD module to separate XSD compilation from genjava which allows for XSD compilation on browser

lchanmann committed:
Fix failing xsd compilation in browser

lchanmann committed:
Support compileAll() on browser

lchanmann committed:
Activate xsd flag when loading XSD module

lchanmann committed:
Merge branch 'development' into Support_xsd_compliation_on_browser

lchanmann committed:
Fix version reference

lchanmann committed:
Include webroot and images directories and deployment/u journals and resources in jar

lchanmann committed:
Add webroot/error.html, needed when starting HttpServer without "nanos.webroot" param

lchanmann committed:
Copy files to webroot

kgrgreer commented on tools/build.js:
```diff
@@ -253,6 +263,38 @@ Implementation-Vendor: ${PROJECT.name}
+  var foambin = `foam-bin-${VERSION}.js`;
+  copyFile('./' + foambin, webroot + '/' + foambin);
+
+  copyFile('./nanopay/src/net/nanopay/ui/unauthorizedAccess/unauthorizedAccess.html', webroot + '/unauthorizedAccess.html');
```
This shouldn't be here since build.js is generic and this is NP specific. This information should be in the POM.

lchanmann replied:
Temporarily added them here until we support before and after tasks https://github.com/kgrgreer/foam3/pull/2856 then we can move it to the appropriate project.

lchanmann committed:
Fix FOAM_ROOT for jar file deployment

lchanmann committed:
Add xsd/pom, fixing foam.xsd.XSDCompiler not being loaded in foam-bin.js

- because the XSD.js was loaded as a project instead of file and the file is not being added to foam.loaded

lchanmann committed:
Store xsd in webroot to allow loading from the browser

lchanmann committed:
Fix LiveScriptBundler genjs call with flags and pom

- Support loading xsd

lchanmann committed:
Support project pom relative to root

lchanmann committed:
Fix xsd compilation on nodejs and browser

lchanmann committed:
Fix xsd compile for jar file and foam-bin deployment

lchanmann committed:
Rename variable and update comment

lchanmann committed:
Fix showing instance name in the build log

lchanmann committed:
Package foam for test and benchmark builds

- test and benchmark run from jar

kgrgreer commented on src/foam/nanos/servlet/VirtualHostRoutingServlet.js:
```diff
@@ -156,6 +156,13 @@ foam.CLASS({
             out.println("<script language=\\"javascript\\" src=\\"" + appConfig.getFoamUrl() + "\\" project=\\"" + appConfig.getPom() + "\\"></script>");
           } else {
             out.println("<script async language=\\"javascript\\" src=\\"/service/liveScriptBundler?");
+
```
The <script> tag on the line above knows the pom from foam.poms, couldn't it just add it to the URL instead of having to expose this information through the appConfig?

lchanmann replied:
Deprecated LiveScriptBundler and remove liveScriptBundle from VirtualHostRoutingServlet https://github.com/kgrgreer/foam3/pull/2855/commits/d750b9595bdc7622bdc133f050fedc0ac846d1eb.

lchanmann committed:
Deprecate LiveScriptBundler and remove -w from the build

lchanmann committed:
Merge branch 'development' into Support_xsd_compliation_on_browser

# Conflicts:
#	tools/build.js
